index.d.ts

```diff
@@ -0,0 +1,2148 @@
+   */
+  auto_link: boolean;
+
+  /**
+   * When present, a separate authentication process will be required through the UI to link this account type.
+   */
+  custom_workflow: boolean;
+}
+
+/**
+ * @see https://www.easypost.com/docs/api/node#carrier-account-field-object
+ */
+export declare interface ICarrierAccountField {
+  /**
+   * Each key in the sub-objects of a CarrierAccount's fields is the name of a settable field
+   */
+  key: string;
+
+  /**
+   * The visibility value is used to control form field types, and is discussed in the CarrierType section
+   */
+  visibility: string;
+
+  /**
+   * The label value is used in form rendering to display a more precise field name
+   */
+  label: string;
+
+  /**
+   * Checkbox fields use "0" and "1" as False and True, all other field types present plaintext, partly-masked, or masked credential data for reference
+   */
+  value: string;
+}
+
+
+/**
+ * A ScanForm can be created to speed up and simplify the carrier pickup process. 
+ * The ScanForm is one document that can be scanned to mark all included tracking codes as "Accepted for Shipment" by the carrier. 
+ * The following criteria must met before creation:
+ *  - Refunded Shipments cannot be added
+ *  - Each Shipment must have the same origin address
+ *  - Shipments must all be dated (using the label_date option) on or after the date the form is generated
+ *  - Shipments cannot be added to more than one ScanForm
+ *  - Existing ScanForms may not be updated with additional Shipments. If a ScanForm already exists, and new Shipments need to be added, a new ScanForm must be created.
+ *  - Shipments should be provided in the form of an array
+ * 
+ * @see https://www.easypost.com/docs/api/node#scan-form-object
+ */
+export declare interface IScanForm extends IObjectWithId<"ScanForm">, IDatedObject {
+  /**
+   * Current status. 
+   * Possible values are "creating", "created" and "failed"
+   */
+  status: "creating" | "created" | "failed";
+
+  /**
+   * Human readable message explaining any failures
+   */
+  message: string;
+
+  /**
+   * Address that the Shipments will be shipped from
+   */
+  address: IAddress;
+
+  /**
+   * Tracking codes included on the ScanForm
+   */
+  tracking_codes: string[];
+
+  /**
+   * Url of the document
+   */
+  form_url: string;
+
+  /**
+   * File format of the document
+   */
+  form_file_type: string;
+
+  /**
+   * The id of the associated Batch. Unique, starts with "batch_"
+   */
+  batch_id: string;
+}
+
+/**
+ * @todo PostageLabel hasn't be documented - https://www.easypost.com/docs/api/node
+ */
+export declare interface IPostageLabel extends IObjectWithId<"PostageLabel">, IDatedObject {
+  date_advance: number;
+  integrated_form: string;
+  label_date: string;
+  label_resolution: number;
+  label_size: string;
+  label_type: string;
+  label_url: string;
+  label_file_type: string;
+  label_pdf_url: string;
+  label_epl2_url: string;
+  label_zpl_url: string;
+}
```

Comment:
I wasn't able to find **PostageLabel** [documented anywhere](https://www.easypost.com/docs/api/node). I was able to capture the properties from an [example response](https://www.easypost.com/docs/api/node#buy-an-order). Can you point me to the documentation for the PostageLabel object?

Reply:
Unfortunately this is an undocumented object (we have plans to add it) but for now, the response example should work.